commands/deploy_folder.js

```diff
@@ -44,39 +45,78 @@ exports.builder = {
     }
 }
 
+
+
 exports.handler = async function (argv) {
 
     list_files_dir(argv.folder, Files)
 
-    generate_seed_and_address_for_every_file(Files, algo, argv.seed, Seed, Address)
 
-    buildTransferTransaction(Address, argv.seed)
+    for (let i = 0; i < Files.length; i++)
+    {
+        Seed.push(generate_seed_and_address(Files[i], algo, argv.seed).seed)
+        Address.push(generate_seed_and_address(Files[i],algo,argv.seed).address)
+    }
+
+    let txn = buildTransferTransaction(Address, argv.seed)
 
     await sendtxn(txn, argv.endpoint)
 
+    async function prepare_transaction (Files)
+    {
+        
+    let seed = generate_seed_and_address(Files, algo, argv.seed).seed
+        
+    let address = generate_seed_and_address(Files, algo, argv.seed).address
+
+    const content = setFileContent(Files)
+
+    let index = await getTransactionIndex(address, argv.endpoint)
+
+    let transaction = buildHostingTransaction(content,seed,index)
+
+    return {transaction,address}
+
+    }
+
     for (let i = 0; i < Files.length; i++) {
        
-        x = Files[i]
-        let seed = generate_seed_and_address(x, algo, argv.seed).seed
-        let address = generate_seed_and_address(x, algo, argv.seed).address
+        let transaction = (await prepare_transaction(Files[i])).transaction
 
+        try {
+          
+            const { fee: fee, rates: rates } = await archethic.getTransactionFee(transaction, argv.endpoint)
+            
+            console.log(chalk.cyan(Files[i]))
 
-        buildHostingTransaction(Files[i])
+            tfee += fee
+            usd += rates.usd
+            eur += rates.eur
 
-        index = await getTransactionIndex(address, argv.endpoint)
+        } catch (e) {
+            console.error(chalk.red(e.message))
+            return
+        }
 
-        buildtxn(seed, index)
+    }
 
-        try {
+    const ok = await yesno({
+        question:  chalk.yellow('The transaction would cost ' +tfee+ ' UCO ($ ' +usd+ ' € ' +eur+ '). Do you want to confirm ?')
+    });
 
-            const ok = await feeconfirmation(transaction, argv.endpoint)
+    for (let i = 0; i < Files.length; i++) {
+       
+        let transaction = (await prepare_transaction(Files[i])).transaction
+
+        let address = (await prepare_transaction(Files[i])).address
```

Comment:
`for` loop executes even after user input is `no` from the user, I think we should evaluate user input `if(ok)` before entering into the loop